lib/cancan/ability.rb

```diff
@@ -61,15 +61,12 @@ module Ability
     #
     # Also see the RSpec Matchers to aid in testing.
     def can?(action, subject, *extra_args)
-      subject = extract_subjects(subject)
-
-      match = subject.map do |subject|
-        relevant_rules_for_match(action, subject).detect do |rule|
-          rule.matches_conditions?(action, subject, extra_args)
+      extract_subjects(subject).each do |a_subject|
```

Comment:
`detect` stops on the first matching result already (http://ruby-doc.org/core-2.3.0/Enumerable.html#method-i-detect)